scripts/benchmarks_local.py

```diff
@@ -0,0 +1,459 @@
+from performance.logger import setup_loggers
+
+# Assumptions: We are only testing this Performance repo, should allow single run or multiple runs
+# For dotnet_version based runs, use the benchmarks_monthly .py script instead
+# Verify the input commands
+# What are supported default cases: MonoJIT, MonoAOTLLVM, MonoInterpreter, Corerun, etc. (WASM)
+
+start_time = datetime.now()
+local_shared_string = "local"
+
+class RunType(Enum):
+    CoreRun = 1
+    MonoAOTLLVM = 2
+    MonoInterpreter = 3
+    MonoJIT = 4
+
+def is_windows(parsed_args: Namespace):
+    return parsed_args.os == "windows"
+
+def is_running_as_admin(parsed_args: Namespace):
+    if is_windows(parsed_args):
+        return ctypes.windll.shell32.IsUserAnAdmin()
+    else:
+        return os.getuid() == 0
+
+def kill_dotnet_processes(parsed_args: Namespace):
+    if is_windows(parsed_args):
+        os.system('TASKKILL /F /T /IM dotnet.exe 2> nul || TASKKILL /F /T /IM VSTest.Console.exe 2> nul || TASKKILL /F /T /IM msbuild.exe 2> nul || TASKKILL /F /T /IM ".NET Host" 2> nul')
+    else:
+        os.system('killall -9 dotnet 2> /dev/null || killall -9 VSTest.Console 2> /dev/null || killall -9 msbuild 2> /dev/null || killall -9 ".NET Host" 2> /dev/null') # Always kill dotnet so it isn't left with handles on its files
+
+def enum_name_to_enum(EnumType, enum_name: str):
+    for enum in EnumType:
+        if enum.name == enum_name:
+            return enum
+    raise ValueError(f"Enum name {enum_name} not found in {EnumType}.")
+
+def enum_name_list_to_enum_list(EnumType, enum_name_list: list):
+    return [enum_name_to_enum(EnumType, enum_name) for enum_name in enum_name_list]
+
+def check_for_runtype_specified(parsed_args: Namespace, run_types_to_check: list) -> bool:
+    for run_type in run_types_to_check:
+        if run_type.name in parsed_args.run_type_names:
+            return True
+    return False
+
+# Uses python copy, to copy the contents of a directory to another directory while overwriting any existing files
+def copy_directory_contents(src_dir: str, dest_dir: str):
+    for src_dirpath, src_dirnames, src_filenames in os.walk(src_dir):
+        dest_dirpath = os.path.join(dest_dir, os.path.relpath(src_dirpath, src_dir))
+        if not os.path.exists(dest_dirpath):
+            os.makedirs(dest_dirpath)
+        for src_filename in src_filenames:
+            if os.path.exists(os.path.join(dest_dirpath, src_filename)) and os.path.samefile(os.path.join(src_dirpath, src_filename), os.path.join(dest_dirpath, src_filename)):
+                continue
+            shutil.copy2(os.path.join(src_dirpath, src_filename), dest_dirpath)
+        
+# Builds libs and corerun by default
+def build_runtime_dependency(parsed_args: Namespace, repo_path: str, subset: str = "clr+libs", configuration: str = "Release", additional_args: list = []):    
+    # Run the command
+    if is_windows(parsed_args):
+        build_libs_and_corerun_command = [
+                "pwsh",
```

Comment:
Is _pwsh_ needed. Maybe people don't have it installed. Maybe regular _powershell_ would do.